streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1030,8 +1038,11 @@ private static Metrics getMetrics(final StreamsConfig config, final Time time, f
                     streamThread.shutdown();
                     threads.remove(streamThread);
                     final long cacheSizePerThread = getCacheSizePerThread(getNumLiveStreamThreads());
+                    final long maxBufferSizePerThread = getBufferSizePerThread(getNumLiveStreamThreads());
                     log.info("Resizing thread cache due to terminating added thread, new cache size per thread is {}", cacheSizePerThread);
```

Comment:
The phrasing here is a little awkward, can we try to clean it up a bit here and in the new line below? (Also worth considering to combine into a single log line here, although in this case imo it's fine either way)
```suggestion
                    log.info("Resizing thread cache again since new thread can not be started, final cache size per thread is {}", cacheSizePerThread);
```